crypto/txscript/src/lib.rs

```diff
@@ -512,6 +536,21 @@ impl<'a, T: VerifiableTransaction, Reused: SigHashReusedValues> TxScriptEngine<'
     }
 }
 
+trait SpkEncoding {
+    fn to_bytes(&self) -> Vec<u8>;
+}
+
+impl SpkEncoding for ScriptPublicKey {
+    fn to_bytes(&self) -> Vec<u8> {
+        let version = self.version.to_be_bytes();
+        let script = self.script();
+        let mut v = Vec::with_capacity(version.len() + script.len());
+        v.extend_from_slice(&version);
+        v.extend_from_slice(script);
+        v
```

Comment:
Any particular reason not to use the style suggested below ? 
We do it often in many places and it has the same prealloc semantics since the iterators are all `ExactSizeIterator` 

```suggestion
        self.version.to_be_bytes().into_iter().chain(self.script().iter().copied()).collect()
```